nslsii/EPS_devices/valves_and_shutters.py

```diff
@@ -0,0 +1,339 @@
+                'the kwargs ``state1_str`` or ``state2_str``')
+
+        self._cmd_map = {self._state1_str: self._state1_cmd,
+                         self._state2_str: self._state2_cmd}
+        self._target_map = {self._state1_str: self._state1_val,
+                            self._state2_str: self._state2_val}
+
+        self._time_fmtstr = '%Y-%m-%d %H:%M:%S'
+
+    _state1_cmd = FmtCpt(EpicsSignal,
+                         '{self.prefix}Cmd:{self._state1_pv_uid}-Cmd',
+                         string=True, kind='omitted')
+    _state2_cmd = FmtCpt(EpicsSignal,
+                         '{self.prefix}Cmd:{self._state2_pv_uid}-Cmd',
+                         string=True, kind='omitted')
+
+    status = Cpt(EpicsSignalRO, 'Pos-Sts', string=True)
+
+    enabled_status = Cpt(EpicsSignalRO, 'Enbl-Sts', string=True, kind='config')
+
+    _fail_to_state1 = FmtCpt(EpicsSignalRO,
+                             '{self.prefix}Sts:Fail{self._state1_pv_uid}-Sts',
+                             string=True, kind='omitted')
+    _fail_to_state2 = FmtCpt(EpicsSignalRO,
+                             '{self.prefix}Sts:Fail{self._state2_pv_uid}-Sts',
+                             string=True, kind='omitted')
+
+    def set(self, val):
+        if self._set_st is not None:
+            raise RuntimeError(
+                f'trying to set {self.name} while another set is in progress')
+
+        cmd_sig = self._cmd_map[val]
+        target_val = self._target_map[val]
+
+        st = self._set_st = DeviceStatus(self)
+        enums = self.status.enum_strs
+
+        def state_cb(value, timestamp, **kwargs):
+            value = enums[int(value)]
+            if value == target_val:
+                st._finished()
+                self._set_st = None
+                self.status.clear_sub(state_cb)
+
+        cmd_enums = cmd_sig.enum_strs
+        count = 0
+
+        def cmd_retry_cb(value, timestamp, **kwargs):
+            nonlocal count
+            value = cmd_enums[int(value)]
+            count += 1
+            if count > self._num_retries:
+                cmd_sig.clear_sub(cmd_retry_cb)
+                self._set_st = None
+                self.status.clear_sub(state_cb)
+                st._finished(success=False)
+            if value == 'None':
+                if not st.done:
+                    time.sleep(self._retry_sleep_time)
+                    cmd_sig.set(1)
+                    ts = datetime.datetime.fromtimestamp(
+                        timestamp).strftime(self._time_fmtstr)
+                    print(f'** ({ts}) Had to reactuate {self.name} while'
+                          f'moving to {val}')
+                else:
+                    cmd_sig.clear_sub(cmd_retry_cb)
+
+        cmd_sig.subscribe(cmd_retry_cb, run=False)
+        cmd_sig.set(1)
+        self.status.subscribe(state_cb)
+
+        return st
+
+    def stop():
+        self.status.clear_sub(state_cb)
+        self._set_st._finished(succsess=False)
```

Comment:
```suggestion
        self._set_st._finished(success=False)
```

Reply:
If I remember correctly, the stop method requires `success` keyword, which should be passed to the line 164 (see my suggestion above).

Reply:
@mrakitin, good catch, I have made the change

Reply:
Thanks @awalter-bnl! I think the `success` kwarg should still be added, as it's done in https://github.com/NSLS-II/nslsii/blob/0ff403eb24b5a1acb79f93e434a7415b06fefb2b/nslsii/devices.py#L88

Reply:
thanks @mrakitin I have made this change in an upcoming commit